modules/client-common/src/main/java/org/apache/ignite/client/proto/ClientMessageUnpacker.java

```diff
@@ -584,6 +606,92 @@ public Object unpackObject(int dataType) {
         throw new IgniteException("Unknown client data type: " + dataType);
     }
 
+    /**
+     * Packs an object.
+     *
+     * @return Object array.
+     * @throws IllegalStateException in case of unexpected value type.
+     */
+    public Object[] unpackObjectArray() {
+        assert refCnt > 0 : "Unpacker is closed";
+
+        if (tryUnpackNil())
+            return null;
+
+        int size = unpackArrayHeader();
+
+        if (size == 0)
+            return ArrayUtils.OBJECT_EMPTY_ARRAY;
+
+        Object[] args = new Object[size];
+
+        for (int i = 0; i < size; i++) {
+            MessageFormat format = getNextFormat();
+
+            switch (format) {
+                case NIL:
+                    unpackNil();
+
+                    break;
+                case BOOLEAN:
+                    args[i] = unpackBoolean();
+
+                    break;
+                case FLOAT32:
+                    args[i] = unpackFloat();
+
+                    break;
+                case FLOAT64:
+                    args[i] = unpackDouble();
+
+                    break;
+                case POSFIXINT:
+                    args[i] = extractExtendedValue(unpackInt());
```

Comment:
I'm not sure we can correctly restore numeric type here.

As I understand @ptupitsyn concerns in one of my PR:
when we serializes small Long value (i.e. which is fit to short) and it can be wrote as short value, so it will be deserialized to Short.

This may lead to cast error in case `Long x = (Long)arr[i]`, if `arr[i]` is a Short.

Reply:
@ptupitsyn, could you, please, chim in?

Reply:
@AMashenkov `extractExtendedValue` solves that problem. `POSFIXINT` means that there is a type code before the actual value.

Reply:
@vladErmakov07 what is a bit confusing though is that we write type code only for some of the types. I understand that some Ignite types map 1-to-1 to MsgPack types, but maybe it'll be cleaner and more future-proof if we always have a type code.

Reply:
@pavlukhin I refactored packObjectArray method and added type code for every type. Now I use unpackObject method to unpack objects.

Reply:
I believe a wrong mention, is not it?